src/tqdm_publisher/_demos/_parallel_bars/_server.py

```diff
@@ -0,0 +1,221 @@
+    [0.4 for _ in range(10)],
+    [0.5 for _ in range(10)],
+]
+
+WEBSOCKETS = {}
+
+## NOTE: TQDMProgressHandler cannot be called from a process...so we just use a queue directly
+progress_handler = TQDMProgressHandler()
+
+
+def forward_updates_over_sse(request_id, id, n, total, **kwargs):
+    progress_handler._announce(dict(request_id=request_id, id=id, format_dict=dict(n=n, total=total)))
+
+
+class ThreadedHTTPServer:
+    def __init__(self, port: int, callback):
+        self.port = port
+        self.callback = callback
+
+    def run(self):
+        create_http_server(port=self.port, callback=self.callback)
+
+    def start(self):
+        thread = threading.Thread(target=self.run)
+        thread.start()
+
+
+def forward_to_http_server(url: str, request_id: str, id: int, n: int, total: int, **kwargs):
+    """
+    This is the parallel callback definition.
+    Its parameters are attributes of a tqdm instance and their values are what a typical default tqdm printout
+    to console would contain (update step `n` out of `total` iterations).
+    """
+    json_data = json.dumps(obj=dict(request_id=request_id, id=str(id), data=dict(n=n, total=total)))
+
+    requests.post(url=url, data=json_data, headers={"Content-Type": "application/json"})
+
+
+def _run_sleep_tasks_in_subprocess(
+    args,
+    # task_times: List[float], iteration_index: int, id: int, url: str
+):
+    """
+    Run a 'task' that takes a certain amount of time to run on each worker.
+
+    In this case that task is simply time.sleep.
+
+    Parameters
+    ----------
+    sleep_time : float
+        The amount of time this task emulates having taken to complete.
+    iteration_index : int
+        The index of this task in the list of all tasks from the buffer map.
+        Each index would map to a different tqdm position.
+    id : int
+        Identifier of ??.
+    url : str
+        The localhost URL to sent progress updates to.
+    """
+
+    task_times, iteration_index, request_id, url = args
+
+    id = uuid.uuid4()
+
+    sub_progress_bar = TQDMPublisher(
+        iterable=task_times,
+        position=iteration_index + 1,
+        desc=f"Progress on iteration {iteration_index} ({id})",
+        leave=False,
+    )
+
+    sub_progress_bar.subscribe(lambda format_dict: forward_to_http_server(url, request_id, id, **format_dict))
+
+    for sleep_time in sub_progress_bar:
+        time.sleep(sleep_time)
+
+
+def run_parallel_processes(request_id, url: str):
+
+    with ProcessPoolExecutor(max_workers=N_JOBS) as executor:
+
+        # Assign the parallel jobs
+        job_map = executor.map(
+            _run_sleep_tasks_in_subprocess,
+            [(task_times, iteration_index, request_id, url) for iteration_index, task_times in enumerate(TASK_TIMES)],
+        )
+
+        # Send initialization for pool progress bar
+        forward_to_http_server(url, request_id, id=request_id, n=0, total=len(TASK_TIMES))
+
+        for _ in job_map:
```

Comment:
This is the line you'd probably have to wrap with the global TQDM - fairly sure that as it performed the iterations in the job_map (deployment to subprocesses) it will increment as each job completes

Reply:
When I used `job_map`, it only updated twice instead of all five times—resulting in an incomplete bar.

Reply:
Might need to switch to the `as_completed` strategy then using a `futures: list` that appends each `executor.submit(...)`

Any questions on that or would you like me to do it

Reply:
Would be great if you can handle it. I'll convert when done

Reply:
My attempts to wrap this line with `TQDMPublisher` (using `job_map` or not) fail to deliver any updates at all

However, I did just push a version that works for me doing the manual calls to the forwarding commands - can you tinker with this more to try to see where the issue is? It just is acting as if the `update` call of the TQDM applied to the executor never gets triggered, but it definitely is in the console

Reply:
Fixed! Just some minor mismatches.